test/libsolidity/util/TestFileParser.cpp

```diff
@@ -199,13 +203,20 @@ pair<string, bool> TestFileParser::parseFunctionSignature()
 	return {signature, !hasName};
 }
 
-u256 TestFileParser::parseFunctionCallValue()
+TestFileParser::FunctionValue TestFileParser::parseFunctionCallValue()
 {
 	try
 	{
-		u256 value{parseDecimalNumber()};
-		expect(Token::Ether);
-		return value;
+		u256 value{ parseDecimalNumber() };
+		auto token = m_scanner.currentToken();
+		auto coin = token == Token::Wei ? FunctionCallValueCoin::Wei : FunctionCallValueCoin::Ether;
+
+		if (token != Token::Ether && token != Token::Wei)
```

Comment:
Maybe better to have this check before assigning to `coin`.

Reply:
Done